gen/bind_generator.jl

```diff
@@ -80,7 +78,35 @@ It is assumed that the HDF library names are given in global constants named `li
 and `libhdf5_hl`. The former is used for all `ccall`s, except if the C library name begins
 with "H5DO" or "H5TB" then the latter library is used.
 """
-macro bind(sig::Expr, err::Union{String,Expr,Nothing} = nothing)
+macro bind(sig::Expr, err::Union{String,Expr,Nothing} = nothing,
+           vers::Union{Expr,Nothing} = nothing)
+    expr = _bind(__module__, __source__, sig, err)
```

Comment:
What's the rationale for  passing  __module, __source__ in this context?

Reply:
Nvm, I just read the comment regarding `__source__`  lower in the file.

Reply:
The `__source__` is used just before the end of the function. I included `__module__` just because it makes the function call basically equivalent to the original macro definition.

Reply:
Albeit we strip the line number anyways, but they are probably helpful for debugging?  (no objections, just nothing this)

Reply:
Yeah, in the grand scheme, we still strip away that line node when writing out `api.jl`, but in principle the macro maintains some source information if you're to use it interactively.

Reply:
Although it looks like we end up spitting the same line number information twice.  

E.g.
```
#= api_defs.jl:22 =#

function h5_close()
    #= api_defs.jl:22 =#
    var"#status#" = ccall((:H5close, libhdf5), herr_t, ())
    var"#status#" < 0 && error("Error closing the HDF5 resources")
    return nothing
end
```

I'm not sure where that first line-number information is coming from.

Reply:
I think the first one is added during parsing of the `api_defs.jl`-contents String:
```julia
julia> defs = read("gen/api_defs.jl", String);

julia> exprs = Meta.parseall(defs, filename = "api_defs.jl"); # called within Base.include_string

julia> exprs.head = :block; # from :toplevel to :block to get nicer printing

julia> exprs
...
    #= api_defs.jl:307 =#
    #= api_defs.jl:307 =# @bind h5tb_read_table(loc_id::hid_t, table_name::Ptr{UInt8}, dst_size::Csize_t, dst_offset::Ptr{Csize_t}, dst_sizes::Ptr{Csize_t}, dst_buf::Ptr{Cvoid})::herr_t "Error reading table"
    #= api_defs.jl:308 =#
    #= api_defs.jl:308 =# @bind h5tb_write_records(loc_id::hid_t, table_name::Ptr{UInt8}, start::hsize_t, nrecords::hsize_t, type_size::Csize_t, field_offsets::Ptr{UInt8}, field_sizes::Ptr{UInt8}, data::Ptr{Cvoid})::herr_t "Error writing record to table"
    #= api_defs.jl:314 =#
    #= api_defs.jl:314 =# @bind h5z_register(filter_class::Ref{H5Z_class_t})::herr_t "Unable to register new filter"
end
```